event/consumer.go

```diff
@@ -46,35 +46,29 @@ func Consume(ctx context.Context, messageConsumer kafka.IConsumerGroup, handler
 }
 
 // processMessage unmarshals the provided kafka message into an event and calls the handler.
-// After the message is handled, it is committed.
-func processMessage(ctx context.Context, message kafka.Message, handler Handler, cfg *config.Config) {
+// After the message is handled, it is committed, by default even on error to prevent reconsumption
+// of dead messages.
+func processMessage(ctx context.Context, msg kafka.Message, h Handler, cfg *config.Config) error {
```

Comment:
If the Consume function doesn't Commit and Release (only Release), then this function would be responsible to commit the message. 

I originally created the auto-generated code to give responsability to this function so that we can decide what errors we want to retry, but I think this is might be confusing, and probably not required by the retry logic that Nathan is designing.

The clearest solution might be to leave `processMessage` as it is and do `CommitAndRelease` in `Consume()`

Reply:
I added the Commit() to a defer as as it stood it was being commited in every case.. My reasoning was if we need to add logic for whether to commit or not we could add a `commit` bool or or something similar

Reply:
sorry I didn't see the defer!